File: src/main/python/TestFb.py
#!/usr/bin/env -s poetry run python

import os
import sco3fb.Hello as Hello
import flatbuffers

PYTHON_FB = "/tmp/hello-python.fb"
JAVA_FB = "/tmp/hello-java.fb"

bld = flatbuffers.Builder(1024)
name = bld.CreateString("Hello")
Hello.Start(bld)
Hello.AddSize(bld, 314)
Hello.AddName(bld, name)
offset: int = Hello.End(bld)
h = bld.Finish(offset)
buf = bld.Output()

with open(PYTHON_FB, "wb") as out_file:
    out_file.write(buf)

if os.path.exists(JAVA_FB):
    with open(JAVA_FB, "rb") as in_file:
        bytes = in_file.read()
        h = Hello.Hello.GetRootAs(bytes, 0)
        print(h.Name().decode("utf-8"), h.Size())

File: src/main/java/sco3/TestFb.java
package sco3;

import static java.nio.file.Paths.get;

import java.io.IOException;
import java.nio.file.Files;

import com.google.flatbuffers.FlatBufferBuilder;

import sco3fb.Hello;

public class TestFb {
	private static final String JAVA_FB_FILE = "/tmp/hello-java.fb";
	private static final String PYTHON_FB_FILE = "/tmp/hello-python.fb";

	public static void main(String[] argv) throws IOException {
		writeToFile();
		readFromFile();
		readFromPythonFile();
	}

	private static void readFromFile() throws IOException {
//		byte[] b = readAllBytes(get(JAVA_FB_FILE));
//		Hello aHello = Hello.parseFrom(b);
//		out.println("Data from file: " + JAVA_FB_FILE + " -> [\n" + aHello.getName() + "]");
	}

	private static void readFromPythonFile() throws IOException {
//		if (Files.exists(get(PYTHON_FB_FILE))) {
//			byte[] b = readAllBytes(get(PYTHON_FB_FILE));
//			Hello aHello = Hello.parseFrom(b);
//			out.println("Data from python file: " + JAVA_FB_FILE + " -> [\n" + aHello + "]");
//		} else {
//			out.println("Python file does not exist: " + PYTHON_FB_FILE);
//		}
	}

	private static void writeToFile() throws IOException {
		FlatBufferBuilder bld = new FlatBufferBuilder(1024);
		int offset = bld.createString("Hello");
		Hello.startHello(bld);
		Hello.addName(bld, offset);
		Hello.addSize(bld, 314);
		offset = Hello.endHello(bld);
		bld.finish(offset);
		byte[] buf = bld.sizedByteArray();
		Files.write(get(JAVA_FB_FILE), buf);

	}

}
